### DISCUSSION TITLE
KubeVela + OpenKruise  cloneset 、sidecarset

### DISCUSSION THREAD

**Discussion by skyshaokai** [kubevela/kubevela#3638] [Q&A] [answered]
刚接触KubeVela，请问KubeVela 可以使用 OpenKruise  cloneset 、sidecarset 吗？
如果可以辛苦给一下相关文档。
如果不可以，应该做哪些工作才能支持呢？

**Comment by wangyikewxgm** [comment, marked as answer]
当然可以，现在已经有一了一个实验性质的 kruise 的 addon，你可以添加试验性仓库之后，启用这个addon https://github.com/oam-dev/catalog/blob/master/experimental/addons/README.md 。 这个addon 已经有了支持 clonset 的 component definition https://github.com/oam-dev/catalog/blob/master/experimental/addons/kruise/definitions/cloneset.yaml ，我理解你也可以把sidecarset 做成一个 traitDefinition 支持进来。
